src/functional.cc

```diff
@@ -0,0 +1,42 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "astshim/functional.h"
+#include "astshim/UnitMap.h"
+
+namespace ast {
+
+std::shared_ptr<FrameSet> prepend(FrameSet const& second, FrameSet const& first) {
+    auto const joinDimensions = first.getFrame(FrameSet::CURRENT)->getNaxes();
+    auto const iCurrent = first.getNframe() + second.getCurrent();
+
+    // not const b/c RVO only works if type matches return type exactly
```

Comment:
That makes sense (though it seems like premature optimization to worry about copying a shared pointer). Please expand the comment for `merged` or delete it.

Please don't add a comment about `second` unless you really think it necessary.